src/ray/object_manager/pull_manager.cc

```diff
@@ -461,15 +461,18 @@ void PullManager::TryToMakeObjectLocal(const ObjectID &object_id) {
   }
 
   // If we can restore directly from this raylet, then try to do so.
-  std::string spilled_url = get_locally_spilled_object_url_(object_id);
+  const std::string locally_spilled_url = get_locally_spilled_object_url_(object_id);
   bool can_restore_directly =
-      !spilled_url.empty() ||  // If the object is spilled locally
+      !locally_spilled_url.empty() ||  // If the object is spilled locally
       (!request.spilled_url.empty() &&
        request.spilled_node_id
            .IsNil());  // Or if the object is spilled on external storages.
   if (can_restore_directly) {
+    const auto spilled_url =
```

Comment:
Would `const auto & spilled_url ` work?

Reply:
i think so if we don't do funky things with `restore_spilled_object_`